docs/contents/dev-docs/api-inference.md

```diff
@@ -0,0 +1,542 @@
+# API Inference
+
+This document discusses how ACK introspects an AWS API model file and
+determines which `CustomResourceDefinition`s (CRDs) to construct and what the
+structure of those CRDs look like.
+
+## The Kubernetes Resource Model
+
+The [Kubernetes Resource Model][krm] (KRM) is a set of [standards][api-stds]
+and naming conventions that govern how an [`Object`][object] may be created and
+updated.
+
+[krm]: https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/
+[api-stds]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md
+[object]: https://kubernetes.io/docs/reference/glossary/?all=true#term-object
+
+An `Object` includes some metadata about the object -- a
+[`GroupVersionKind`][gvk] (GVK), a `Name`, a `Namespace`, and zero or more `Labels`
+and `Annotations`.
+
+[gvk]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#resources
+
+In addition to this metadata, each `Object` has a `Spec` field which is a
+struct that contains the **desired** state of the `Object`. `Objects` are
+typically denoted using YAML, like so:
+
+```yaml
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  name: my-amazing-bucket
+  annotations:
+    pronounced-as/boo-kay
```

Comment:
nit: [Annotations are key/value maps](https://github.com/kubernetes/apimachinery/blob/15c5dba13c59f42b25aba4e3a6b17061af770216/pkg/apis/meta/v1/types.go#L233-L238), did you mean:
```yaml
annotations:
  - pronounced-as: boo-kay
```

Reply:
lol, thanks for catching this!

Reply:
fixed.